### PR TITLE
Migrate test framework

### PR DESCRIPTION
**Pull request by satoshi-szk**
Hi, @yuichkun. I set up jest. Could you check please?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "prepublish": "yarn run build"
   },
   "dependencies": {
+    "babel-core": "7.0.0-bridge.0",
```
Comment (yuichkun) [inline package.json]: Not that big of a problem, but this should probably be under devDependencies.

Comment (satoshi-szk) [reply package.json]: Yeah, I’ll fix.

File: package.json [hunk 2]
```diff
@@ -7,8 +7,9 @@
     "lib": "lib"
   },
   "scripts": {
-    "test": "jest",
     "pretest": "yarn run build",
+    "test": "jest test/integration",
+    "test:unit": "jest test/unit",
```
Comment (yuichkun) [inline package.json]: I have one suggestion: 
Jest's watch mode is pretty powerful, so how about we make this `jest test/unit --watch`?

**Review by yuichkun — APPROVED** [review]
Overall PR looks pretty good to me. Types properly shows up in my editor, tests run without errors.

**Pull request merged**